tasks/git/src/main/java/com/walmartlabs/concord/plugins/git/GitHubTask.java

```diff
@@ -114,92 +119,91 @@ public GitHubTask(boolean dryRunMode) {
-                return getLatestSHA(in, gitHubUri);
-            }
-            case CREATEREPO: {
-                return createRepo(in, gitHubUri);
-            }
-            case DELETEREPO: {
-                return deleteRepo(in, gitHubUri);
-            }
-            case GETCONTENT: {
-                return getContent(in, gitHubUri);
-            }
-            case CREATEHOOK: {
-                return createHook(in, gitHubUri);
-            }
-            case GETPRFILES: {
-                return getPRFiles(in, gitHubUri);
-            }
-            default:
-                throw new IllegalArgumentException("Unsupported action type: " + action);
+    public Map<String, Object> execute(Map<String, Object> in, Map<String, Object> defaults, GitSecretService secretService) {
+        Action action = getAction(in);
+        GitHubApiInfo apiInfo = createApiInfo(in, defaults, secretService);
+
+        log.info("Starting '{}' action on API URL: {}", action, apiInfo.baseUrl());
+
+        return switch (action) {
+            case CREATEPR -> createPR(in, apiInfo);
+            case COMMENTPR -> commentPR(in, apiInfo);
+            case GETPRCOMMITLIST -> getPRCommitList(in, apiInfo);
+            case MERGEPR -> mergePR(in, apiInfo);
+            case CLOSEPR -> closePR(in, apiInfo);
+            case CREATEISSUE -> createIssue(in, apiInfo);
+            case CREATETAG -> createTag(in, apiInfo);
+            case DELETETAG -> deleteTag(in, apiInfo);
+            case DELETEBRANCH -> deleteBranch(in, apiInfo);
+            case MERGE -> merge(in, apiInfo);
+            case GETCOMMIT -> getCommit(in, apiInfo);
+            case ADDSTATUS -> addStatus(in, apiInfo);
+            case GETSTATUSES -> getStatuses(in, apiInfo);
+            case FORKREPO -> forkRepo(in, apiInfo);
+            case GETBRANCHLIST -> getBranchList(in, apiInfo);
+            case GETPR -> getPR(in, apiInfo);
+            case GETPRLIST -> getPRList(in, apiInfo);
+            case GETTAGLIST -> getTagList(in, apiInfo);
+            case GETLATESTSHA -> getLatestSHA(in, apiInfo);
+            case CREATEREPO -> createRepo(in, apiInfo);
+            case DELETEREPO -> deleteRepo(in, apiInfo);
+            case GETCONTENT -> getContent(in, apiInfo);
+            case CREATEHOOK -> createHook(in, apiInfo);
+            case GETPRFILES -> getPRFiles(in, apiInfo);
+            case CREATEAPPTOKEN -> createAppToken(apiInfo);
+        };
+    }
+
+    protected AccessTokenProvider getTokenProvider(Map<String, Object> in, String baseUrl, GitSecretService secretService) {
+        String owner = assertString(in, GITHUB_ORGNAME);
+        String repo = assertString(in, GITHUB_REPONAME);
+
+        Map<String, Map<String, Object>> auth = getMap(in, "auth", Map.of());
+        String token = getString(in, GITHUB_AUTH_ACCESSTOKEN, null);
+
+        if (auth.isEmpty()) {
+            log.warn("Using deprecated 'accessToken' input. Please use 'auth.accessToken.token' instead.");
```

Comment:
Are there any other properties in `auth.accessToken` besides `.token`? Why not just `auth.accessToken`?

Reply:
There is not other expected properties. Treating the `auth` value as a Map makes it a bit easier to handle it similarly to the other auth types and run it through the same convertValue call and then use with the [returned `AccessTokenProvider`](https://github.com/walmartlabs/concord-plugins/pull/203/files/75cc43d9c64031e369eb0181db29566564f3634b#diff-325d4948fa2e02fa8871e662159ab2cc150cafd454dc6da242c40d51a1d048d4R203). Otherwise we need more routes of logic handle just a plain ol' `String` value.

Reply:
That's a fair approach. Addressed in latest commit.